driver/src/main/scala/api/collections/UpdateOps.scala

```diff
@@ -170,22 +170,23 @@ trait UpdateOps[P <: SerializationPack] extends UpdateCommand[P]
      * }
      * }}}
      */
-    final def many(updates: Iterable[UpdateElement])(implicit ec: ExecutionContext): Future[MultiBulkWriteResult] = updates.headOption match {
-      case Some(first) => {
+    final def many(updates: Iterable[UpdateElement])(implicit ec: ExecutionContext): Future[MultiBulkWriteResult] = {
+      if (updates.nonEmpty) {
         val bulkProducer = BulkOps.bulks(
-          updates, maxBsonSize, metadata.maxBulkSize) { up =>
-          elementEnvelopeSize + pack.bsonSize(up.q) + pack.bsonSize(up.u)
+          updates, maxBsonSize, metadata.maxBulkSize) {
+          up =>
+            elementEnvelopeSize + pack.bsonSize(up.q) + pack.bsonSize(up.u)
         }
 
         BulkOps.bulkApply[UpdateElement, UpdateWriteResult](
-          bulkProducer)({ bulk =>
-          execute(first, bulk.drop(1).toSeq)
```

Comment:
Here is the bug: we override the first element of the bulk by the first element of the sequence.

Reply:
The `first` element to `updates` is the first to be executed as bulk.
Do not see your point.

Reply:
The point is the first element of the second bulk of updates will never be written to mongo, the same for the third bulk etc. The code `drops` the first element of bulks, assuming that the first element is passed as the first argument to `execute`, which is not true.
If the explanation isn't clear yet, please see the added test. The test creates 2 elements of equal size, writes them to mongo and checks the number of modified elements. The elements are so big that they need to be written in 2 bulks. Without the fix, i.e. the second commit, the test fails.